determined-ai/determined
# refactor: removes subscription modification and replaces with soft resets

## Description
Removes code paths for subscription modifications and replaces with ability to send multiple startup messages to perform a soft reset on an existing subscription while maintaining the original connection.





## Test Plan
- Run `stream.py` and subscribed to any trial
- update the trial that `stream.py` is connected to at least twice
- this should trigger `late_sub()` to run and you'll enter a new subscription 
- confirm that the new subscription was successful and you only receive messages relevant to the new subscription





## Commentary (optional)





## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.

## Ticket
DET-9808
